src/plugins/mf_priority.cpp

```diff
@@ -191,19 +199,76 @@ static int validate_cb (flux_plugin_t *p,
         return flux_jobtap_reject_job (p, args, "unable to unpack bank arg");
     }
 
-    // searching element in std::map by key
+    // make sure user belongs to flux-accounting DB
     it = users.find (userid);
     if (it == users.end ())
         return flux_jobtap_reject_job (p, args,
                                        "user not found in flux-accounting DB");
 
+    // make sure user belongs to bank they specified; if no bank was passed in,
+    // look up their default bank
     if (bank != NULL) {
-        inner_it = it->second.find (std::string (bank));
-        if (inner_it == it->second.end ())
+        bank_it = it->second.find (std::string (bank));
+        if (bank_it == it->second.end ())
             return flux_jobtap_reject_job (p, args,
-                                    "user does not belong to specified bank");
+                                     "user does not belong to specified bank");
+    } else {
+        bank = const_cast<char*> (users_def_bank[userid].c_str ());
     }
 
+    try {
+        max_jobs = users.at (userid).at (bank).max_jobs;
+        current_jobs = users.at (userid).at (bank).current_jobs;
+    } catch (const std::out_of_range& oor) {
+        return flux_jobtap_reject_job (p, args,
+                                 "could not look up user's active job count");
+    }
+
+    if (flux_jobtap_job_aux_set (p,
+                                 FLUX_JOBTAP_CURRENT_JOB,
+                                 "mf_priority:bank",
+                                 bank,
+                                 NULL) < 0)
+        flux_log_error (h, "flux_jobtap_job_aux_set");
+
+    // make sure user has not already hit their max active jobs count
+    if (current_jobs >= max_jobs)
+        return flux_jobtap_reject_job (p, args,
+                                       "user has max number of jobs submitted");
+
+    users[userid][bank].current_jobs++;
+
+    return 0;
+}
+
+
+static int inactive_cb (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *data)
+{
+    int userid;
+    char *bank;
+
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i}",
+                                "userid", &userid) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_unpack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
+    bank = static_cast<char *> (flux_jobtap_job_aux_get (
+                                                    p,
+                                                    FLUX_JOBTAP_CURRENT_JOB,
+                                                    "mf_priority:bank"));
+
+    users[userid][bank].current_jobs--;
```

Comment:
same question about catching any exception on this line (honestly above my head though)